tests/pyfunc/test_inferred_data_path.py

```diff
@@ -0,0 +1,125 @@
+import pickle
+import mlflow
+import os
+from mlflow.models import Model
+import numpy as np
+import pytest
+import sklearn.neighbors
+import sklearn.datasets
+import sys
+from pathlib import Path
+
+sys.path.insert(0, ".")
+
+
+@pytest.fixture
+def model_path(tmp_path):
+    return tmp_path / "model"
+
+
+@pytest.fixture(scope="module")
+def iris_data():
+    iris = sklearn.datasets.load_iris()
+    x = iris.data[:, :2]
+    y = iris.target
+    return x, y
+
+
+@pytest.fixture(scope="module")
+def sklearn_knn_model(iris_data):
+    x, y = iris_data
+    knn_model = sklearn.neighbors.KNeighborsClassifier()
+    knn_model.fit(x, y)
+    return knn_model
+
+
+def _walk_dir(path):
+    for sub_path in path.glob('**/*'):
+        if sub_path.is_file():
+            rel_path = sub_path.relative_to(path)
+            yield str(rel_path)
```

Comment:
```suggestion
def _walk_dir(path):
    return {str(p.relative_to(path)) for p in path.rglob("*") if p.is_file()}
```

Reply:
not sure if `_walk_dir` needs to be an iterator.

Reply:
nice simplification